### ISSUE TITLE
engine.h is missing

### ISSUE THREAD

**Issue by Myron472** [CauldronDevelopmentLLC/cbang#165]
When I was compiling Cbang on my Fedora 41 I got this error:
src/cbang/openssl/KeyPair.cpp:55:10: Fatal Error: openssl/engine.h: No such file or directory

**Comment by jcoffland** [comment]
Your OpenSSL installation is either broken or the wrong version.  Although, I don't know of any OpenSSL release that does not include ``openssl/engine.h``.

**Comment by Myron472** [comment]
Is this `openssl/engine.h` path for cbang's openssl or Fedora's openssl? Because I initially thought that engine.h is missing in the cbang, not in my system.

**Comment by jcoffland** [comment]
It should be on your system.

**Comment by Myron472** [comment]
Ok, I actually figured out what was the problem. I didn't have **openssl-devel-engine** package installed.
For context:
I needed to compile FoldingAtHome v8 for my Fedora system as the developers of this project didn't provide pre-built packages for RedHat-based systems. In the GitHub repo of it there is an instruction for building rpm, and it says to download certain packages to be able to build FAH. But the openssl-devel-engine, as it turned out, is also needed for building FAH. However, the instruction doesn't say to install it.

Anyway, thanks for the response!
